ir_Lego_PF.cpp

```diff
@@ -44,3 +43,55 @@ void IRsend::sendLegoPowerFunctions(uint16_t data, bool repeat)
 }
 
 #endif // SEND_LEGO_PF
+
+//+=============================================================================
+#if DECODE_LEGO_PF
+bool  IRrecv::decodeLegoPowerFunctions (decode_results *results) {
+	static LegoPfBitStreamEncoder bitStreamEncoder;
+	uint16_t data   = 0;
+	int  offset = 1;
+
+	// Check SIZE
+	if (irparams.rawlen < 2 * (bitStreamEncoder.MESSAGE_BITS) - 1)  return false ;  // exclude STOP pause
+
+	// Check START_BIT Mark/Space
+	if (!MATCH_MARK (results->rawbuf[offset++], bitStreamEncoder.IR_MARK_DURATION - 58)) return false ;
+	if (!MATCH_SPACE(results->rawbuf[offset++], bitStreamEncoder.START_PAUSE_DURATION + 100)) return false ;
+
+	while(offset < irparams.rawlen - 1) {
+		if (MATCH_MARK(results->rawbuf[offset], bitStreamEncoder.IR_MARK_DURATION - 58)) {
+			offset++ ;
+		}
+		else {
+			return false ;
+		}
+
+		// ONE & ZERO
+		if (MATCH_SPACE(results->rawbuf[offset], bitStreamEncoder.HIGH_PAUSE_DURATION + 50)) {
+			data = (data << 1) | 1 ;
+		}
+		else if (MATCH_SPACE(results->rawbuf[offset], bitStreamEncoder.LOW_PAUSE_DURATION + 62)) {
+			data = (data << 1) | 0 ;
+		}
+		else {
+			break ;  // End of one & zero detected
+		}
+		offset++;
+	}
+
+	results->bits = (offset - 2) / 2;

```

Comment:
Thanks for the head up. Better checking no of bits found first before setting 16 into results->bits. Fixed.